ostd/libs/linux-bzimage/builder/src/encoder.rs

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! This module is used to compress kernel ELF.
+
+use std::{fs::read_to_string, io::Write};
+
+use libflate::{deflate, gzip, zlib};
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+struct TomlManifest {
+    config: Config,
+}
+
+#[derive(Debug, Deserialize)]
+struct Config {
+    run: RunConfig,
+}
+
+#[derive(Debug, Deserialize)]
+struct RunConfig {
+    build: BuildConfig,
+}
+
+#[derive(Debug, Deserialize)]
+struct BuildConfig {
+    compression_format: String,
+}
+
+pub fn compress_kernel(kernel: &[u8]) -> Vec<u8> {
+    let mut current_dir = std::env::current_dir().unwrap();
+    current_dir.pop();
+    current_dir.push("asterinas/bundle.toml");
```

Comment:
No. The builder should not parse the toml file generated by OSDK, which will make it dependent on OSDK's implementation.

Just define a enum "CompressionFormat" here, and let `compress_kernel` and `make_bzimage` take an extra argument about how to compress.